### ISSUE TITLE
Could spring stream add a annotation "StreamFunction" ?

### ISSUE THREAD

**Issue by frlzk** [spring-cloud/spring-cloud-stream#2247]
Add a annotation StreamFunction, indicate some function is used in stream. Avoid centralized configuration of spring.cloud.function.definition
For example:
`
@Bean
@StreamFunction
public Function<String, String> toUpperCase() {
	return s -> s.toUpperCase();
}

@Bean
@StreamFunction
public Supplier<Date> date() {
	return () -> new Date(12345L);
}
`

**Comment by olegz** [comment]
I wonder what value would it add other then bring us back to where we once were with `@StreamListener` and mixed configuration. The `spring.cloud.function.definition` is just one of the many configuration properties, so even if we did what you propose you would still need to configure the rest externally, hence mixed configuration which is never a good idea as it is hard to follow.
Please clarify. . .
